### PR TITLE
Create templates for virtual hearing reminder emails

### PR DESCRIPTION
**Pull request by ferristseng**
Connected to #15400
Resolves https://vajira.max.gov/browse/CASEFLOW-194

### Description

- Creates reminder email templates for appellant and representative
- Updates the rake task to generate sample emails to work with emails without attachments
- Fixes a factory bug that caused `virtual_hearing.hearing.virtual_hearing` to return `nil`
- Small tweak to how virtual hearing establishment got created so it works when using `build(:virtual_hearing)`
- Add tests for virtual hearing reminder email subject line

#### Appellant Reminder Email

Screen Shot 2020-11-03 at 2 23 59 PM

#### POA Reminder Email

Screen Shot 2020-11-03 at 2 24 39 PM

### CODE REVIEW

File: lib/tasks/emails.rake [hunk 1]
```diff
@@ -34,21 +34,25 @@ namespace :emails do
...
+          :cancellation,
+          :reminder
         ].each do |func|
           email = VirtualHearingMailer.send(
             func,
             mail_recipient: recipient,
             virtual_hearing: hearing.virtual_hearing
           )
+          email_body = if email.body.nil? || email.body.empty?
+                         email.html_part&.decoded
```
Comment (ferristseng) [inline lib/tasks/emails.rake:46]: When I was using this locally, it seemed like `html_part` was only defined when there was an attachment set (for a multipart email?).

File: spec/factories/virtual_hearing.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@
     representative_tz { nil }
     association :created_by, factory: :user
     association :updated_by, factory: :user
-    establishment { nil }
+    establishment { build(:virtual_hearing_establishment) }
```
Comment (ferristseng) [inline spec/factories/virtual_hearing.rb:21]: Small tweak to always build this associated instance. Won't affect the code

File: spec/factories/virtual_hearing.rb [hunk 3]
```diff
@@ -42,11 +42,14 @@
       judge_email_sent { true }
     end
 
+    after(:create) do |virtual_hearing, _evaluator|
+      # Calling reload after create fixes a problem where calling `virtual_hearing.hearing.virtual_hearing`
+      # would return `nil`.
+      virtual_hearing.reload
```
Comment (ferristseng) [inline spec/factories/virtual_hearing.rb:48]: This fixes a bug/quirk that led some of the test fixtures to setup test data using an `update` as opposed to using a variable.

File: spec/mailers/virtual_hearing_mailer_spec.rb [hunk 4]
```diff
@@ -29,7 +39,9 @@
 
   shared_context "legacy_hearing" do
     let(:hearing) do
-      hearing_date = Time.use_zone("America/New_York") { Time.zone.now.change(hour: 11, min: 30) }
+      hearing_date = Time.use_zone("America/New_York") do
+        Time.zone.now.change(hour: 11, min: 30) + 1.day # Tomorrow. Matches the AMA hearing scheduled for.
```
Comment (ferristseng) [inline spec/mailers/virtual_hearing_mailer_spec.rb:43]: Change this to use the same date as the AMA hearing. To make it more explicit, I declared `Date.tomorrow` in the `hearing_day` despite it being the default.

File: app/mailers/virtual_hearing_mailer.rb [hunk 5]
```diff
@@ -8,6 +8,7 @@ class VirtualHearingMailer < ActionMailer::Base
   helper VirtualHearings::CalendarTemplateHelper
 
   def cancellation(mail_recipient:, virtual_hearing: nil)
+    # Guard to prevent cancellation emails from sending to the judge
```
Comment (sahalliburton) [inline app/mailers/virtual_hearing_mailer.rb:11]: Thanks for the comment, this is super helpful!

**Review by sahalliburton — APPROVED** [review]
LGTM, thanks for all the tests!

**Pull request merged**